Automattic/wp-calypso
# Jetpack connect: store pre-selected plan in cookie

Store any plan selected before connecting a site in a cookie instead of redux state. This means that the plan will persist across both a login and the redirect for authorising the site. 

Redux state does not persist across logins and is not guaranteed to persist across redirects away from calypso (depends on the timing).

This PR and #20067 will combine to fix pre-selecting plans when logged-out, which currently does not work, making the customer need to select the plan twice.

## Testing

This can be tested both logged-in and logged-out, ~though only logged-in will use the saved plan until #20067 lands~ ✅ .

1. Go to /jetpack/connect/store
1. Choose a plan
1. Connect a site
1. Should be sent to the cart after site auth (logged-in)
